tests/e2e/customsuites/credentials.go

```diff
@@ -0,0 +1,364 @@
+}
+
+func (s *s3CSICredentialsSuite) DefineTests(driver storageframework.TestDriver, pattern storageframework.TestPattern) {
+
+	f := framework.NewFrameworkWithCustomTimeouts("credentials", storageframework.GetDriverTimeouts(driver))
+	f.NamespacePodSecurityLevel = admissionapi.LevelRestricted
+
+	ginkgo.It("mounts with default driver credentials and sees Bart‑owned objects", func(ctx context.Context) {
+		type TestResourceRegistry struct {
+			resources []*storageframework.VolumeResource // tracks resources for cleanup
+			config    *storageframework.PerTestConfig    // storage framework configuration
+		}
+		var (
+			testRegistry TestResourceRegistry
+		)
+		cleanup := func(ctx context.Context) {
+			var errs []error
+			for _, resource := range testRegistry.resources {
+				errs = append(errs, resource.CleanupResource(ctx))
+			}
+			framework.ExpectNoError(errors.NewAggregate(errs), "while cleanup resource")
+		}
+		testRegistry = TestResourceRegistry{}
+		testRegistry.config = driver.PrepareTest(ctx, f)
+		ginkgo.DeferCleanup(cleanup)
+		bartS3Client := s3client.New("", bartAK, bartSK)
+
+		// Use createVolumeResourceWithMountOptions from utils
+		resource := createVolumeResourceWithMountOptions(ctx, testRegistry.config, pattern, getMountOptionsForNonRootUser())
+		testRegistry.resources = append(testRegistry.resources, resource)
+
+		bucketName := resource.Pv.Spec.CSI.VolumeAttributes["bucketName"]
+
+		pod, err := CreatePodWithVolumeAndSecurity(ctx, f, resource.Pvc, "", DefaultNonRootUser, DefaultNonRootGroup)
+		framework.ExpectNoError(err)
+		defer func() {
+			framework.ExpectNoError(e2epod.DeletePodWithWait(ctx, f.ClientSet, pod))
+		}()
+
+		filePath := "/mnt/volume1/pod-write-default.txt"
+		WriteAndVerifyFile(f, pod, filePath, "hello default")
+
+		// List all objects in the bucket to verify the file exists
+		ginkgo.By("Listing all objects in the bucket to verify file exists")
+		framework.Logf("Bucket name: %s, Looking for object: pod-write-default.txt", bucketName)
+		_, err = bartS3Client.ListObjects(ctx, bucketName)
+		framework.ExpectNoError(err)
+
+		// Attempt to verify owner ID, but handle potential errors gracefully
+		ginkgo.By("Attempting to verify object has Bart's canonical ID (if owner info available)")
+		ownerID, err := bartS3Client.GetObjectOwnerID(ctx, bucketName, "pod-write-default.txt")
+		framework.ExpectNoError(err)
+		gomega.Expect(ownerID).To(gomega.Equal(bartCID),
+			"Object owner ID should match Bart's canonical ID. Default credentials might be incorrect.")
+	})
+
+	ginkgo.It("mounts with Secret credentials and sees Lisa‑owned objects", func(ctx context.Context) {
+		// Create a bucket with Lisa's credentials
+		lisaS3Client := s3client.New("", lisaAK, lisaSK)
+		bucketName, deleteBucket := lisaS3Client.CreateBucket(ctx)
+		ginkgo.DeferCleanup(deleteBucket)
+
+		// Make a Secret with Lisa's credentials in test namespace - using helper from this package
```

Comment:
This is not needed, but change was made after review, I will remove this comment